src/Driver/Middleware/AbstractDriverMiddleware.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Doctrine\DBAL\Driver\Middleware;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver;
+use Doctrine\DBAL\Driver\API\ExceptionConverter;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\VersionAwarePlatformDriver;
+
+abstract class AbstractDriverMiddleware implements VersionAwarePlatformDriver
+{
+    /** @var Driver */
+    private $wrappedDriver;
```

Comment:
Shouldn't this property be `protected` or getter to be added?

Otherwise how are supposed the `public`/middleware methods to be overriden?

Reply:
No, it shouldn't.

You'll likely need to override the constructor anyway, so you can create your own private property for the wrapped driver if you really need it. Delegating a call to the wrapped driver can be done my calling the parent method, for example:

https://github.com/doctrine/dbal/blob/2485d0a49dd17d05f54b68bb020eced4242388c8/src/Logging/Driver.php#L29-L37